### ISSUE TITLE
Capabilities

### ISSUE THREAD

**Issue by machi1990** [graphqlcrud/graphqlcrud-java#16]
- Pagination
- CRUD operations
- Filterings
- Ordering
- Relationships

**Comment by machi1990** [comment]
I tested out the current master https://github.com/graphqlcrud/graphqlcrud-java/commit/fa8438ed900df91a08d39386b80fc4d3e01e97e8, filtering seems to working (I'll perform extensive tests later on), but good job so far.  

I have a question regarding the list query type e.g `accounts` from the shared image, are there plans to support pagination, i.e instead of returning the raw type but a container type that has raw type items and pagination info as indicated in https://graphqlcrud.org/docs/next/find#example-query

See: 
![app-testing](https://user-images.githubusercontent.com/10106536/93062621-828b5f80-f675-11ea-9407-fc75d1c61364.png)
 
/cc @rareddy @wtrocki @ani-sha

**Comment by ani-sha** [comment]
Yes, we have thought of adding pagination after modifying and implementing the filters entirely. There are a few more tests and functions to add.

**Comment by rareddy** [comment]
In the pagination scenario, what mechanism do you offer the user to get the "next" page after the first page is returned?

**Comment by wtrocki** [comment]
In dynamic systems using calculated cursor would be more appropriate. 
We use limit and offset, but that is susceptible to seeing duplicates etc.

**Comment by wtrocki** [comment]
On the client side pages will be kept in the component state and deduplicated so we could provide support for two types of pagination:

- Unlimited scrolling (uses the client-side state to fetch the next page without calculating count)
- Client requests count and uses limit offset for `classical` pagination


While this has obvious challenges this is something we have implemented and we have flexibility to change `offset` to be used as generic string `token` to provide backward compatibility if we hit issues.

**Comment by rareddy** [comment]
ok, yes with offset you can see the potential for incorrect results when database state changed, but that is easiest otherwise we would need to build a local cache for it :(

How is paging done at the nested level? or this only at the top level?

**Comment by wtrocki** [comment]
Pagination and filtering can be done on the nested level as well.
